lib/src/css_mirroring/rulesets_processor.dart

```diff
@@ -0,0 +1,82 @@
+part of scissors.src.css_mirroring.bidi_css_generator;
+
+abstract class NodeProcessor {
+  void process(EditConfiguration e, TopLevelsPair t, int i);
+}
+
+class DeclarationProcessor extends NodeProcessor {
+  final _PendingRemovals removableRuleSets;
+  final int parentEnd;
+
+  DeclarationProcessor(this.removableRuleSets, this.parentEnd);
+
+  @override
+  void process(EditConfiguration _editConfig, TopLevelsPair _topLevelsPair,
+      int _iTopLevel) {
+    final removals = new _PendingRemovals(_editConfig.transaction);
+    final List<TreeNode> usedTopLevels =
```

Comment:
well, it sounds like you're getting top levels that have been used, so it's confusing.
I'd name the function topLevelsForMode, for instance, but this is still weird: there's an assumption that removableRuleSets operates on the same toplevels as that "used" toplevels? (surely there should be a way to make this clearer)